core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -988,6 +988,12 @@ private[spark] class BlockManager(
         logWarning(s"Putting block $blockId failed")
       }
       res
+    } catch {
+      // Since removeBlockInternal may throw exception,
+      // we should print exception first to show root cause.
+      case e: Throwable =>
+        logWarning(s"Putting block $blockId failed due to exception $e.")
```

Comment:
Will we see the message `Putting block $blockId failed due to exception` twice in a log file?

Reply:
Update

Reply:
Would you please change to `case NonFatal(e) =>`

Reply:
Done @jerryshao